### ISSUE TITLE
Check all objectives for NaN gradients

### ISSUE THREAD

**Issue by f0uriest** [PlasmaControl/DESC#738]
In several places we inadventently divide by zero, but then replace the value with something sensible. This works fine in forward mode AD, but can cause issues in reverse mode (like `jax.grad`)
